sympy/physics/mechanics/inertia.py

```diff
@@ -0,0 +1,195 @@
+        raise TypeError('Need to define the inertia in a frame')
+    ixx = sympify(ixx)
+    ixy = sympify(ixy)
+    iyy = sympify(iyy)
+    iyz = sympify(iyz)
+    izx = sympify(izx)
+    izz = sympify(izz)
+    ol = ixx * (frame.x | frame.x)
+    ol += ixy * (frame.x | frame.y)
+    ol += izx * (frame.x | frame.z)
+    ol += ixy * (frame.y | frame.x)
+    ol += iyy * (frame.y | frame.y)
+    ol += iyz * (frame.y | frame.z)
+    ol += izx * (frame.z | frame.x)
+    ol += iyz * (frame.z | frame.y)
+    ol += izz * (frame.z | frame.z)
+    return ol
+
+
+def inertia_of_point_mass(mass, pos_vec, frame):
+    """Inertia dyadic of a point mass relative to point O.
+
+    Parameters
+    ==========
+
+    mass : Sympifyable
+        Mass of the point mass
+    pos_vec : Vector
+        Position from point O to point mass
+    frame : ReferenceFrame
+        Reference frame to express the dyadic in
+
+    Examples
+    ========
+
+    >>> from sympy import symbols
+    >>> from sympy.physics.mechanics import ReferenceFrame, inertia_of_point_mass
+    >>> N = ReferenceFrame('N')
+    >>> r, m = symbols('r m')
+    >>> px = r * N.x
+    >>> inertia_of_point_mass(m, px, N)
+    m*r**2*(N.y|N.y) + m*r**2*(N.z|N.z)
+
+    """
+
+    return mass * (((frame.x | frame.x) + (frame.y | frame.y) +
+                    (frame.z | frame.z)) * (pos_vec & pos_vec) -
+                   (pos_vec | pos_vec))
+
+
+class Inertia:
+    """Inertia object consisting of a Dyadic and a Point of reference.
+
+    Explanation
+    ===========
+
+    This is a simple class to store the Point and Dyadic, belonging to an
+    inertia.
+
+    Attributes
+    ==========
+
+    dyadic : Dyadic
+        The dyadic of the inertia.
+    point : Point
+        The reference point of the inertia.
+
+    Examples
+    ========
+
+    >>> from sympy.physics.mechanics import ReferenceFrame, Point, Inertia
+    >>> N = ReferenceFrame('N')
+    >>> Po = Point('Po')
+    >>> Inertia(N.x.outer(N.x) + N.y.outer(N.y) + N.z.outer(N.z), Po)
+    Inertia((N.x|N.x) + (N.y|N.y) + (N.z|N.z), Po)
+
+    In the example above the Dyadic was created manually, one can however also
+    use the ``inertia`` function for this or the class method ``from_tensor`` as
+    shown below.
+
+    >>> Inertia.from_tensor(Po, N, 1, 1, 1)
+    Inertia((N.x|N.x) + (N.y|N.y) + (N.z|N.z), Po)
+
+    """
+    def __init__(self, dyadic, point):
+        # Switch order if given in the wrong order
+        if isinstance(dyadic, Point) and isinstance(point, Dyadic):
+            point, dyadic = dyadic, point
+        if not isinstance(point, Point):
+            raise TypeError('Reference point should be of type Point')
+        if not isinstance(dyadic, Dyadic):
+            raise TypeError('Inertia value should be expressed as a Dyadic')
+        self._dyadic = dyadic
+        self._point = point
+
+    @classmethod
+    def from_tensor(cls, point, frame, ixx, iyy, izz, ixy=0, iyz=0, izx=0):
```

Comment:
This feels like a lot of positional arguments to me. Others may be happy with it though. My main concern would be about getting the order of `point` and `frame` correct.

Reply:
This mimics existing api, so at least people would be used to it.

The most confusion I've seen on order is when we have:

```python
def f(a_tuple, something, else=None):
```
where
```python
a_tuple = (thing1, thing2)
```

When you type `help(f)` or `f?` the order of (thing1, thing2) hasn't been obvious from the first handful of lines from the docstring, thus you forget the order. This happened with the orient api `.orient(frame_A, 'Axis', (q, A.x))` or was it `(A.x, q)`. We ended up just supporting both orders, which was actually a good idea. But when we moved to `orient_axis()` this is now clean and the order is clear in the function sig. The function signature here will print first and it shows the point frame order in the first line, so I think we are ok.

The other issue with this that I've hit is that you forget if you should give `ixz` or `izx`. Supporting them both would relieve you of that detail.

Reply:
I agree that it would in a way be a nice feature to support `ixz` as well as `izx`. However you'll also have to handle also cases that someone parses both. This increases complexity and source code for not that good of a reason, a proper IDE shows the arguments a function needs. Besides supporting both would contradict the Python zen rule: _There should be one-- and preferably only one --obvious way to do it._

Reply:
No need to support that, agreed, but sometimes breaking that guideline is beneficial (was in the orient() code). Also ask the core Python devs how many ways there are to do string formatting :smile: